scripts/fetch-common-contracts.js

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env node
+
+// This script snapshots the bytecode and ABI for the `hardhat/common-contracts.js` script.
+// - Bytecode is fetched directly from the blockchain by querying the provided client endpoint. If not endpoint is
+//   provided, ethers default provider is used instead.
+// - ABI is fetched from etherscan's API using the provided etherscan API key. If no api key is provided, ABI will not
+//   be fetched and saved.
+//
+// The produced artifacts are stored in the `output` folder ('test/bin' by default). For each contract, two files are
+// produced:
+// - `<name>.bytecode` containing the contract bytecode (in binary encoding)
+// - `<name>.abi` containing the ABI (in utf-8 encoding)
+
+const fs = require('fs');
+const path = require('path');
+const { ethers } = require('ethers');
+const { request } = require('undici');
+
+const { argv } = require('yargs/yargs')()
```

Comment:
```suggestion
const { argv } = require('yargs/yargs')(process.argv.slice(2))
```
Don't want to allow passing them as params?

Reply:
I though Yargs would do it automatically. Added that using the hideBin utils that is more precise. 
`.slice(2)` would not work if calling the file directly (it is `chmod+x` with the `#!`)